src/contact.rs

```diff
@@ -620,7 +621,7 @@ impl Contact {
     pub async fn get_all(
         context: &Context,
         listflags: u32,
-        query: Option<impl AsRef<str>>,
+        query: Option<&str>,
```

Comment:
I also removed `AsRef` to avoid specifying the type for `None` in tests.

Generally, `AsRef` is not what you want unless you are designing some very generic API. If someone calls the function with `str` and someone calls the function with `String`, you get two functions compiled into binary which only differ in what type they accept, and the binary size is increased unnecessarily.